x/ecocredit/server/basket/query_balance.go

```diff
@@ -0,0 +1,21 @@
+package basket
+
+import (
+	"context"
+
+	baskettypes "github.com/regen-network/regen-ledger/x/ecocredit/basket"
+)
+
+func (k Keeper) BasketBalance(ctx context.Context, request *baskettypes.QueryBasketBalanceRequest) (*baskettypes.QueryBasketBalanceResponse, error) {
```

Comment:
should we be checking if the request is nil?

Reply:
i'll add. @amaurym can you confirm whether this is needed or not in grpc?